src/Accord.Bot/Infrastructure/BotState.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Threading;
+
+namespace Accord.Bot.Infrastructure
+{
+    public class BotState
+    {
+        private long _isCacheReady;
+        public bool IsCacheReady
+        {
+            get => Interlocked.Read(ref _isCacheReady) == 1;
+            set => Interlocked.Exchange(ref _isCacheReady, Convert.ToInt64(value));
+        }
```

Comment:
Is it necessary to use interlocked here, I can only see this being set in the ready responder?

Reply:
The previous revision was waiting in a loop for cache to be ready. Now it just discards any command if the cache isn't ready.
With the infinite loop it was necessary, now it probably isn't. So i can remove it

Reply:
Yeah would be good to remove it